src/Utilities/functions.jl

```diff
@@ -946,10 +955,6 @@ end
     filter_variables(keep::Function, f::AbstractFunction)
 
 Return a new function `f` with the variable `vi` such that `!keep(vi)` removed.
-
-WARNING: Don't define `filter_variables(::Function, ...)` because Julia will
```

Comment:
When did this change?

Reply:
It hasn't. I don't think this change was intentional. It's unrelated to Nonlinear and not present in `master`. I'll revert.